### PR TITLE
Fixes MM-50733

### PR DESCRIPTION
**Pull request by srkgupta**
#### Summary
https://mattermost.atlassian.net/browse/MM-50733

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/oauth.go [hunk 1]
```diff
@@ -309,6 +309,10 @@ func (a *App) GetOAuthAccessTokenForCodeFlow(clientId, grantType, redirectURI, c
 			return nil, model.NewAppError("GetOAuthAccessToken", "api.oauth.get_access_token.internal_user.app_error", nil, "", http.StatusNotFound)
 		}
 
+		if user.DeleteAt != 0 {
```
Comment (m1lt0n) [inline server/channels/app/oauth.go:312]: (question) out of curiosity: is this temporary code for handling the users that have oauth data but have been deactivated? or will this check remain as an extra check?

File: server/channels/app/user.go [hunk 2]
```diff
@@ -933,6 +933,10 @@ func (a *App) userDeactivated(c request.CTX, userID string) *model.AppError {
 		a.disableUserBots(c, userID)
 	}
 
+	if nErr := a.Srv().Store().OAuth().RemoveAuthDataByUserId(userID); nErr != nil {
```
Comment (m1lt0n) [inline server/channels/app/user.go:936]: Should an extra test be added in the `UpdateUserActive` method's tests to cover that branch?

**Review by m1lt0n — APPROVED** [review]
Nice one! Only one main comment about potentially an extra test due to the conditional added in `userDeactivated`.

File: server/channels/app/oauth.go (hunk 1 above)
Comment (srkgupta) [reply server/channels/app/oauth.go:312]: No, this is to handle any edge case where the `RemoveAuthDataByUserId` errors out in `userDeactivated` or if the user is deactivated without going through the `userDeactivated` flow.

**Review by BenCookie95 — APPROVED** [review]

**Review by oh6hay — APPROVED** [review]

**Comment by m1lt0n** [comment]
@DHaussermann adding for QA review

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Tested issue and was unable to reproduce
- Regression tested CRUD for oath app
- No issues found

LGTM!

Tested issue

**Comment by amyblais** [comment]
@hanzei This needs to be released to cloud first. After it's been released to cloud it can be cherry-picked to self-hosted versions.

**Comment by hanzei** [comment]
:ack: @amyblais Can you please let me know when we are ready to cherry-pick the change into the self-hosted release branches?

**Comment by amyblais** [comment]
Yes 👍
